app/scripts/services/homePagePreference.js

```diff
@@ -0,0 +1,131 @@
+'use strict';
+
+angular.module("openshiftConsole")
+  .service("HomePagePreference", function($location, $timeout, $q, $uibModal, AuthService, Logger, Navigate, NotificationsService, ProjectsService){
+
+    var HOME_PAGE_KEY_PREFIX = "openshift/home-page-pref/";
+
+    function getHomePagePreferenceKey() {
+      var user = AuthService.getUser();
+      if (!user) {
+        return null;
+      }
+      return HOME_PAGE_KEY_PREFIX + user.metadata.name;
+    }
+
+    var getHomePagePreference = function() {
+      var homePagePref;
+      var key = getHomePagePreferenceKey();
+
+      if (!key) {
+        return 'catalog-home';
+      }
+
+      try {
+        homePagePref = JSON.parse(localStorage.getItem(key));
+      } catch (e) {
+        Logger.error('Could not parse homePagePref as JSON');
+        return 'catalog-home';
+      }
+
+      return _.get(homePagePref, 'type', 'catalog-home');
+    };
+
+    var getProjectOverviewPage = function() {
+      var homePagePref;
+      var key = getHomePagePreferenceKey();
+
+      if (!key) {
+        return null;
+      }
+
+      try {
+        homePagePref = JSON.parse(localStorage.getItem(key));
+      } catch (e) {
+        Logger.error('Could not parse homePagePref as JSON');
+        return null;
+      }
+
+      return (homePagePref && homePagePref.type === 'project-overview') ? homePagePref.project : null;
+    };
+
+    var verifyProjectOverviewPage = function(projectOverviewPage) {
+      var deferred = $q.defer();
+
+      ProjectsService.get(projectOverviewPage).then(function() {
+        deferred.resolve(true);
+      }, function() {
+        deferred.resolve(false);
+      });
+
+      return deferred.promise;
+    };
+
+    var clear = function() {
+      var key = getHomePagePreferenceKey();
+      if (!key) {
+        return;
+      }
+      localStorage.removeItem(key);
+    };
+
+    var notifyInvalidProjectHomePage = function(projectName) {
+      clear();
```

Comment:
This will `clear` your preference on any error. So if you lose network connectivity or something and the project is fine, it still clears your preference, which it shouldn't.